code/game/turfs/simulated.dm

```diff
@@ -71,6 +71,17 @@
 							step(H, H.dir)
 							to_chat(H, "<span class='notice'>You tripped over your hair!</span>")
 
+			//A billowing floor-length cloak is a tripping hazard
+			if(H.alphas.len > 0)
```

Comment:
Didn't actually think of the fact that looping that much whenever a mob moves around would happen.

My suggestion would be to do an event hook into the `on_moved` event of mobs:
```dm
/obj/item/weapon/cloakingcloak
    var/event_key

/obj/item/weapon/cloakingcloak/proc/mob_moved(var/list/event_args, var/mob/holder)
    ...

// To register the event:
event_key = mob.on_moved.Add(src, "mob_moved")

// To unregister:
mob.on_moved.Remove(event_key)
event_key = null
```